lib/Backend/arm64/LowerMD.cpp

```diff
@@ -1353,89 +1341,95 @@ LowererMD::LowerEntryInstr(IR::EntryInstr * entryInstr)
-        instrPush->SetSrc1(IR::RegBVOpnd::New(frameRegs, TyMachReg, this->m_func));
-        insertInstr->InsertBefore(instrPush);
+        // Generate STP {fp,lr}
+        IR::Instr * instrStp = IR::Instr::New(Js::OpCode::STP_PRE, this->m_func);
+        instrStp->SetDst(IR::IndirOpnd::New(IR::RegOpnd::New(nullptr, RegSP, TyMachReg, this->m_func), -(regPushedSize + (2 * MachRegInt)), TyMachReg, this->m_func));
+        instrStp->SetSrc1(IR::RegOpnd::New(nullptr, RegFP, TyMachReg, this->m_func));
+        instrStp->SetSrc2(IR::RegOpnd::New(nullptr, RegLR, TyMachReg, this->m_func));
+        insertInstr->InsertBefore(instrStp);
 
         // Generate MOV fp,sp
-        IR::RegOpnd* spOpnd = IR::RegOpnd::New(nullptr, RegSP, TyMachReg, this->m_func);
-        IR::RegOpnd* r11Opnd = IR::RegOpnd::New(nullptr, RegFP, TyMachReg, this->m_func);
-        IR::Instr * instrMov = IR::Instr::New(Js::OpCode::MOV, r11Opnd, spOpnd, this->m_func);
+        IR::Instr * instrMov = IR::Instr::New(Js::OpCode::ADD,
+            IR::RegOpnd::New(nullptr, RegFP, TyMachReg, this->m_func),
+            IR::RegOpnd::New(nullptr, RegSP, TyMachReg, this->m_func),
+            IR::IntConstOpnd::New(0, TyMachReg, this->m_func), this->m_func);
         insertInstr->InsertBefore(instrMov);
+
+        regPushedSize = 0; // STP_PRE Advanced the SP
     }
 
     if (!usedRegs.IsEmpty())
     {
-        // Generate PUSH {r4-r10,r12}
-        IR::Instr * instrPush = IR::Instr::New(Js::OpCode::STP, this->m_func);
-        instrPush->SetDst(IR::IndirOpnd::New(IR::RegOpnd::New(nullptr, RegSP, TyMachReg, this->m_func), (int32)0, TyMachReg, this->m_func));
-        instrPush->SetSrc1(IR::RegBVOpnd::New(usedRegs, TyMachReg, this->m_func));
-        insertInstr->InsertBefore(instrPush);
+        for (RegNum reg = LAST_CALLEE_SAVED_GP_REG; reg > LAST_INT_ARG_REG; reg = (RegNum)(reg - 1))
+        {
+            if (usedRegs.Test(RegEncode[reg]))
+            {
+                regPushedSize += MachRegInt;
+                IR::Instr * instrStore = IR::Instr::New(Js::OpCode::STR, this->m_func);
+                instrStore->SetDst(IR::IndirOpnd::New(IR::RegOpnd::New(nullptr, RegSP, TyMachReg, this->m_func), -regPushedSize, TyMachReg, this->m_func));
+                instrStore->SetSrc1(IR::RegOpnd::New(reg, TyMachReg, this->m_func));
+                insertInstr->InsertBefore(instrStore);
+            }
+        }
     }
 
     if (!usedDoubleRegs.IsEmpty())
     {
-        // Generate VPUSH {d8-d15}
-        IR::Instr * instrPush = IR::Instr::New(Js::OpCode::STP, this->m_func);
-        instrPush->SetDst(IR::IndirOpnd::New(IR::RegOpnd::New(nullptr, RegSP, TyMachReg, this->m_func), (int32)0, TyMachReg, this->m_func));
-        instrPush->SetSrc1(IR::RegBVOpnd::New(usedDoubleRegs, TyMachReg, this->m_func));
-        insertInstr->InsertBefore(instrPush);
+        for (RegNum reg = LAST_CALLEE_SAVED_DBL_REG; reg >= FIRST_CALLEE_SAVED_DBL_REG; reg = (RegNum)(reg - 1))
```

Comment:
You need to use FSTR here. Also floating point registers can be stored in pairs as well, so pairwise logic should eventually be added to use FSTP.

Reply:
Changed to FSTR

---
In reply to: [143538894](https://github.com/Microsoft/ChakraCore/pull/3903#discussion_r143538894) [](ancestors = 143538894)